app/services/selection/selection.ts

```diff
@@ -43,26 +41,74 @@ export class SelectionService
 
   // SELECTION METHODS
 
-  add(itemIds: string | string[]) {
-    const ids: string[] = Array.isArray(itemIds) ? itemIds : [itemIds];
-    this.select(this.state.selectedIds.concat(ids));
-    this.SET_LAST_SELECTED_ID(ids[ids.length - 1]);
+  add: (items: TItemsList) => ISelection;
+  deselect: (items: TItemsList) => ISelection;
+  reset: () => ISelection;
+  selectAll: () => ISelection;
+  invert: () => ISelection;
+  getItems: () => SceneItem[];
+  getVisualItems: () => SceneItem[];
+  getIds: () => string[];
+  getInvertedIds: () => string[];
+  getInverted: () => SceneItem[];
+  getBoundingRect: () => IRectangle;
+  getLastSelected: () => SceneItem;
+  getSize: () => number;
+  isSelected: (item: string | ISceneItem) => boolean;
+  copyReferenceTo: (sceneId: string) => SceneItem[];
+  copyTo: (sceneId: string) => SceneItem[];
+  moveTo: (sceneId: string) => SceneItem[];
```

Comment:
What's the purpose of declaring all these methods here?  They're defined below, and are all declared additionally in `ISelection`.

Reply:
SelectionService use Selection class as a mixin. This is the way how typescript documentation recommends to use mixins. Unfortunately, it requires this additional boilerplate code. https://www.typescriptlang.org/docs/handbook/mixins.html

Reply:
Ah, got it.